src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -363,6 +363,57 @@ private static void AddIsECMABoundaryHelper(Dictionary<string, string[]> require
             }
         }
 
+        /// <summary>Adds an IndexOfAnyValues instance declaration to the required helpers collection if the chars are ASCII.</summary>
+        private static string EmitIndexOfAnyValuesOrLiteral(ReadOnlySpan<char> chars, Dictionary<string, string[]> requiredHelpers)
+        {
+            // IndexOfAnyValues<char> is faster than a regular IndexOfAny("abcd") for sets of 4/5 values iff they are ASCII.
+            // Only emit IndexOfAnyValues instances when we know they'll be faster to avoid increasing the startup cost too much.
+            Debug.Assert(chars.Length is 4 or 5);
+
+            return RegexCharClass.IsAscii(chars)
+                ? EmitIndexOfAnyValues(chars.ToArray(), requiredHelpers)
+                : Literal(chars.ToString());
+        }
+
+        /// <summary>Adds an IndexOfAnyValues instance declaration to the required helpers collection.</summary>
+        private static string EmitIndexOfAnyValues(char[] asciiChars, Dictionary<string, string[]> requiredHelpers)
+        {
+            Debug.Assert(RegexCharClass.IsAscii(asciiChars));
+
+            // The set of ASCII characters can be represented as a 128-bit bitmap. Use the 16-byte hex string as the key.
+            byte[] bitmap = new byte[16];
+            foreach (char c in asciiChars)
+            {
+                bitmap[c >> 3] |= (byte)(1 << (c & 7));
+            }
+
+            string hexBitmap = BitConverter.ToString(bitmap).Replace("-", string.Empty);
+
+            string fieldName = hexBitmap switch
+            {
+                "0000000000000000FEFFFF07FEFFFF07" => "s_asciiLetters",
+                "000000000000FF03FEFFFF07FEFFFF07" => "s_asciiLettersAndDigits",
+                "000000000000FF037E0000007E000000" => "s_asciiHexDigits",
+                "000000000000FF03000000007E000000" => "s_asciiHexDigitsLower",
+                "000000000000FF037E00000000000000" => "s_asciiHexDigitsUpper",
+                _ => $"s_ascii_{hexBitmap.TrimStart('0')}"
+            };
+
+            string helperName = $"IndexOfAnyValues_{fieldName}";
+
+            if (!requiredHelpers.ContainsKey(helperName))
+            {
+                Array.Sort(asciiChars);
+
+                requiredHelpers.Add(helperName, new string[]
+                {
+                    $"internal static readonly IndexOfAnyValues<char> {fieldName} = IndexOfAnyValues.Create({Literal(new string(asciiChars))});",
```

Comment:
```suggestion
                    $"/// <summary>Cached data to efficiently search for a character in the set {Literal(new string(asciiChars).Replace("&", "&amp;").Replace("<", "&lt;").Replace(">", "&gt;"))}.</summary>",
                    $"internal static readonly IndexOfAnyValues<char> {fieldName} = IndexOfAnyValues.Create({Literal(new string(asciiChars))});",
```
or something like that.  Just a nicety so that:
a) Everything in the Utilities class is consistent with having an XML comment, and
b) When someone hovers over the field at usage sites, they get meaningful IntelliSense.

Note there's another place we use Replace like this to sanitize a string for use in XML.  Might want to put that into a simple helper function.